app/src/main/java/com/concordium/wallet/ui/cis2/TokensViewModel.kt

```diff
@@ -401,35 +399,34 @@ class TokensViewModel(application: Application) : AndroidViewModel(application)
                     tokenIds = commaSeparated
                 )
             tokenData.contractName = cis2TokensMetadata.contractName
-            cis2TokensMetadata.metadata.forEach {
-                loadTokenMetadata(tokenData.contractIndex, tokenData.contractName, it)
+            cis2TokensMetadata.metadata.forEach { metadataItem ->
+                loadTokenMetadata(
+                    tokenToUpdate = tokens.first {
+                        it.token == metadataItem.tokenId
+                                && it.contractIndex == tokenData.contractIndex
+                    },
+                    cis2TokensMetadataItem = metadataItem,
+                )
             }
-            return true
+            // TODO handle incorrect metadata for some tokens.
```

Comment:
What is meant here?
I don't like adding a TODO as it will likely never be resolved.

Reply:
Forgot to push the last commit removing this TODO entry

Reply:
Could you push it then? :upside_down_face:

Reply:
Ahh yes, my bad. :sweat_smile: